engine/src/main/java/org/terasology/engine/subsystem/openvr/OpenVRControllers.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.engine.subsystem.openvr;
+
+import jopenvr.VRControllerState_t;
+import org.joml.Matrix4f;
+import org.terasology.input.ButtonState;
+import org.terasology.input.ControllerDevice;
+import org.terasology.input.InputType;
+import org.terasology.input.device.ControllerAction;
+import org.terasology.rendering.openvrprovider.ControllerListener;
+import org.terasology.rendering.openvrprovider.OpenVRUtil;
+import org.terasology.input.ControllerId;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+public class OpenVRControllers implements ControllerDevice, ControllerListener {
+
+    private Queue<ControllerAction> queuedActions = new ArrayDeque<>();
+
+    /**
+     * Get the controller names provided by this ControllerDevice.
+     * @return the list of controllers names.
+     */
+    @Override
+    public List<String> getControllers() {
+        List<String> ids = new ArrayList<>();
+        ids.add("OpenVR");
+        return ids;
+    }
+
+    /**
+     * Get all queued actions registered since this method was last called.
+     * @return a queue of actions.
+     */
+    @Override
+    public Queue<ControllerAction> getInputQueue() {
```

Comment:
Ok, fair enough. Indeed if there are constraints on the naming of something because it has a relationship with something else let's make it clear in the javadocs or in very visible comments.